src/test/java/io/github/anthorx/parquet/sql/write/JDBCWriterTest.java

```diff
@@ -0,0 +1,119 @@
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.io.IOException;
+import java.sql.SQLException;
+
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+public class JDBCWriterTest {
+
+
+  @Mock
+  private PreparedStatementRecordConsumer mockedRecordConsumer;
+
+  @Mock
+  private RecordConsumerInitializer lazyRecordConsumerInitializer;
+
+  @Mock
+  private SQLParquetReader sqlParquetReader;
+
+  @BeforeEach
+  public void setUp() throws SQLException {
+    when(lazyRecordConsumerInitializer.initialize(anyList())).thenReturn(mockedRecordConsumer);
+  }
+
+  private Record createBasicRecord() {
+    Record record = new Record();
+    record.addField(new RecordField<>("int", 10).addReadConsumer(ReadRecordConsumer::setInt));
+    record.addField(new RecordField<>("string", "stringValue").addReadConsumer(ReadRecordConsumer::setString));
+    return record;
+  }
+
+
+  @Test
+  public void recordsAreInsertedUntilTheEnd() throws IOException, SQLException {
+    when(sqlParquetReader.read())
+        .thenReturn(createBasicRecord(), createBasicRecord(), null);
+
+    JDBCWriter jdbcWriter = spy(new JDBCWriter(lazyRecordConsumerInitializer, sqlParquetReader, 2));
+    jdbcWriter.write();
+
+    verify(mockedRecordConsumer, times(2)).addBatch();
+  }
+
+
+  @Test
+  public void recordFieldsAreConsumed() throws IOException, SQLException {
+    when(sqlParquetReader.read()).thenReturn(createBasicRecord(), null);
+
+    doAnswer(invoc -> {
+      int arg = invoc.getArgument(0);
+      Assert.assertEquals(10, arg);
+      return null;
+    }).when(mockedRecordConsumer).setInt(anyInt());
+
+    doAnswer(invoc -> {
+      String arg = invoc.getArgument(0);
+      Assert.assertEquals("stringValue", arg);
+      return null;
+    }).when(mockedRecordConsumer).setString(anyString());
+
+    JDBCWriter jdbcWriter = new JDBCWriter(lazyRecordConsumerInitializer, sqlParquetReader, 2);
+    jdbcWriter.write();
+
+    verify(mockedRecordConsumer, times(1)).addBatch();
+  }
+
+  @Test
+  public void recordAreInsertedInBatchWhen() throws IOException, SQLException {
+    when(sqlParquetReader.read())
+        .thenReturn(createBasicRecord(), createBasicRecord(), createBasicRecord(), null);
+
+    JDBCWriter jdbcWriter = new JDBCWriter(lazyRecordConsumerInitializer, sqlParquetReader, 2);
+    jdbcWriter.write();
+
+    verify(mockedRecordConsumer, times(2)).executeBatch();
+  }
+
+  @Test
+  public void recordAreInsertedInBatch() throws IOException, SQLException {
+    when(sqlParquetReader.read())
+        .thenReturn(createBasicRecord(), createBasicRecord(), createBasicRecord(), createBasicRecord(), null);
+
+    JDBCWriter jdbcWriter = new JDBCWriter(lazyRecordConsumerInitializer, sqlParquetReader, 2);
```

Comment:
This exact JDBCWriter initialization is done in every test, maybe we can move in the @BeforeEach method and make jdbcWriter class attribute ?